templates/components/status.html.twig

```diff
@@ -1,16 +1,45 @@
-{% macro status(text, options) %}
-    {% set _color = options.color ?? 'green' %}
-    {% set _animated = options.animated ?? true %}
-    {% set _lite = options.lite is defined ? options.lite  : false %}
-    {% set _with_dot = options.with_dot is defined ? options.with_dot  : false %}
-    {% set _extraClass = options.extraClass ?? '' %}
-
-    {%- apply spaceless -%}
-        <span class="status status-{{ _color }} {{ _lite ? 'status-lite' : '' }} {{ _extraClass }}">
-        {% if _with_dot %}
-            <span class="status-dot {{ _animated ? 'status-dot-animated' : '' }}"></span>
-        {% endif %}
-            {{ text }}
-        </span>
-    {%- endapply -%}
-{% endmacro %}
+{% macro status(text, options) %}
+    {% import '@Tabler/components/status.html.twig' as macro %}
+    {% import '@Tabler/includes/utils.html.twig' as utils %}
+
+    {% set _color = options.color ?? 'green' %}
+    {% set _animated = (options.animated ?? true) is same as true %}
+    {% set _lite = (options.lite ?? false) is same as true %}
+    {% if options.with_dot is defined %}
+        {% deprecated 'with_dot option is deprecated, use withDot option instead!' %}
+        {% set _withDot = options.with_dot is same as true %}
+    {% else %}
+        {% set _withDot = (options.withDot ?? false) is same as true %}
+    {% endif %}
+
+    {% set _extraClass = options.extraClass ?? '' %}
+    {% set _attr = options.attr ?? {} %}
+
+    {%- apply spaceless -%}
+        <span
+            class="status status-{{ _color }} {% if _lite %}status-lite{% endif %} {{ _extraClass }}"
+            {{ utils.attr_to_html(_attr) }}
+        >
+            {% if _withDot %}
+                {{ macro.status_dot({animated : _animated}) }}

```

Comment:
Lets clean that up, `_self` always works:
https://twig.symfony.com/doc/3.x/tags/macro.html#importing-macros

Reply:
I don't know why, but I have something somewhere in my brain that told me to be careful with that when used inside embed or other complex import..

I'll change that.

Reply:
And I remember that using _self wasn't possible at all ... We are doing this job for too long 😁